### PR TITLE
Allow interal applications to skip the verify process

### PR DESCRIPTION
**Pull request by doliveirakn**

### CODE REVIEW

File: lib/oauth/controllers/provider_controller.rb [hunk 1]
```diff
@@ -57,11 +57,11 @@ def authorize
           @token = ::RequestToken.find_by_token! params[:oauth_token]
           oauth1_authorize
         else
-          if request.post?
-            @authorizer = OAuth::Provider::Authorizer.new current_user, user_authorizes_token?, params
+          @client_application = ClientApplication.find_by_key! params[:client_id]
+          if request.post? || @client_application.internal?
+            @authorizer = OAuth::Provider::Authorizer.new current_user, (user_authorizes_token?|| @client_application.internal?), params
```
Comment (tvongaza) [inline lib/oauth/controllers/provider_controller.rb]: Missing a space after the "user_authorizes_token?", but shouldn't matter.

File: lib/oauth/controllers/provider_controller.rb [hunk 2]
```diff
@@ -57,11 +57,11 @@ def authorize
           @token = ::RequestToken.find_by_token! params[:oauth_token]
           oauth1_authorize
         else
-          if request.post?
-            @authorizer = OAuth::Provider::Authorizer.new current_user, user_authorizes_token?, params
+          @client_application = ClientApplication.find_by_key! params[:client_id]
```
Comment (hansonywu) [inline lib/oauth/controllers/provider_controller.rb:60]: does find_by_key! raise an exception if it can't find one?

Comment (hansonywu) [inline lib/oauth/controllers/provider_controller.rb:60]: Oh, I guess it was there already.

Comment (doliveirakn) [inline lib/oauth/controllers/provider_controller.rb:60]: yeah, I just moved it around.

**Pull request merged**